src/main/java/io/github/cottonmc/cotton/gui/widget/WTiledSprite.java

```diff
@@ -0,0 +1,55 @@
+package io.github.cottonmc.cotton.gui.widget;
+
+import io.github.cottonmc.cotton.gui.client.ScreenDrawing;
+import net.minecraft.util.Identifier;
+
+// It's a Sprite that can be tiled, but does not support animations
+public class WTiledSprite extends WSprite {
+  public int tileWidth = 20;
+  public int tileHeight = 20;
+
+	WTiledSprite(Identifier image) {
+		super(image);
+	}
+
```

Comment:
There's no need to have this constructor as there's no sensible default value for tiling.

Reply:
default could be the size of the passed texture if we can get that value somehow. otherwise i agree